src/posts/posts.service.spec.ts

```diff
@@ -1,18 +1,47 @@
 import { Test, TestingModule } from '@nestjs/testing';
+import { PostRepository } from './entities/post.repository';
 import { PostsService } from './posts.service';
+import { CreatePostDto } from './dto/create-post.dto';
 
 describe('PostsService', () => {
   let service: PostsService;
+  let repo: PostRepository;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      providers: [PostsService],
+      providers: [
+        PostsService,
+        {
+          provide: PostRepository,
+          useValue: {
+            createPost: jest.fn(() => ({ uuid: 'test id' })),
+          },
+        },
+      ],
     }).compile();
 
     service = module.get<PostsService>(PostsService);
+    repo = module.get<PostRepository>(PostRepository);
   });
 
   it('should be defined', () => {
     expect(service).toBeDefined();
   });
+
+  it('should have createPost method', () => {
+    expect(service).toHaveProperty('createPost');
+  });
+
+  describe('createPost Fn', () => {
```

Comment:
same as [this comment](https://github.com/m3ntorship/pickify-posts/pull/72/files#r637458997)

Reply:
It seems that there is no error case here in posts.controller. As any error will be thrown in the methods inside it by nestjs. Accordingly, error handling tests will be written whenever there is an error might be thrown inside the method under test itself. E.g. inside postsRepository method